core/src/test/scala/unit/kafka/cluster/PartitionTest.scala

```diff
@@ -83,6 +87,47 @@ class PartitionTest {
     replicaManager.shutdown(checkpointHW = false)
   }
 
+  @Test
+  def testMaybeReplaceCurrentWithFutureReplica(): Unit = {
+    val latch = new CountDownLatch(1)
+
```

Comment:
Perhaps just add a comment that we want to test if partition.removeFutureLocalReplica() and partition.maybeReplaceCurrentWithFutureReplica() can run concurrently.